src/main/java/com/yandex/money/api/typeadapters/model/showcase/container/GroupTypeAdapter.java

```diff
@@ -152,8 +153,11 @@ public static JsonArray serialize(Group group, JsonSerializationContext context)
         public static Group deserialize(JsonArray jsonArray, JsonDeserializationContext context) {
             Group.Builder builder = new Group.Builder();
             for (JsonElement item : jsonArray) {
-                builder.addItem((Component) context.deserialize(item, ComponentsTypeProvider
-                        .getClassOfComponentType(getTypeFromJsonElement(item))));
+                Component component = context.deserialize(item, ComponentsTypeProvider
+                        .getClassOfComponentType(getTypeFromJsonElement(item)));
+                if (!(component instanceof Undefined)) {
```

Comment:
Now it is more efficient to compare instance of deserializedItem with Undefined.INSTANCE than to use instanceof operator.